src/components/TicketModal/TicketModal.vue

```diff
@@ -2,175 +2,268 @@
-    const testUsers = ref([{ data: "Mason" }, { data: "George" }]);
-    const testTeams = ref([{ data: "Slugbotics" }, { data: "Bugslotics" }]);
-    const testTags = ref([{ data: "Mechanical" }, { data: "Systems" }]);
-    const testStatuses = ref([
-      { data: "To Do" },
-      { data: "In Progress" },
-      { data: "Done" }
-    ]);
-    const statusSelected = (item: string) => {
-      ticketStatus.value = item;
+
+    const initializeData = () => {
+      if (props.ticketRecord) {
+        doesTicketExist.value = true;
+      }
+      selectedStatus.value.id = -1;
+      ticketObj.value.description = "";
+      ticketObj.value.title = "Title";
+      ticketObj.value.due_date = "";
     };
-    const userSelected = (item: string) => {
-      ticketUser.value = item;
+    const getTeams = async () => {
+      try {
+        teams.value = Object.values(await getUsersTeams());
+        selectedTeam.value.name = teams.value[0].name;
+        selectedTeam.value.id = teams.value[0].id;
+      } catch (error) {
+        alert(error);
+      }
     };
-    const teamSelected = (item: string) => {
-      ticketTeam.value = item;
+    const getTags = async () => {
+      try {
+        tags.value = Object.values(await listTags(selectedTeam.value.id));
+        selectedTag.value.title = "None";
+        selectedTagId.value = [-1];
+      } catch (error) {
+        alert(error);
+      }
     };
-    const tagSelected = (item: string) => {
-      ticketTag.value = item;
+    const getMembers = async () => {
+      try {
+        members.value = await listMembersDepaginated(selectedTeam.value.id);
+        selectedUser.value.username = "None";
+        selectedMember.value.id = "-1";
+      } catch (error) {
+        alert(error);
+      }
     };
-    const closeModal = () => {
-      context.emit("close");
+    const getStatuses = async () => {
+      try {
+        statuses.value = Object.values(
+          await listStatuses(selectedTeam.value.id)
+        );
+        selectedStatus.value.title = statuses.value[0].title;
+        selectedStatus.value.color = statuses.value[0].color;
+        selectedStatus.value.id = statuses.value[0].id;
+      } catch (error) {
+        alert(error);
+      }
+    };
+    const statusSelected = (display: string, item: StatusRecordOutput) => {
+      selectedStatus.value.title = display;
+      selectedStatus.value.color = item.color;
+      selectedStatus.value.id = item.id;
+    };
+    const userSelected = (display: string, item: MemberRecord) => {
+      selectedUser.value.username = display;
+      selectedUser.value.pk = item.id;
+      selectedMember.value.id = item.id;
+    };
+    const teamSelected = async (display: string, item: ReadTeamRecord) => {
+      if (selectedTeam.value.id !== item.id) {
+        selectedTeam.value.name = display;
+        selectedTeam.value.id = item.id;
+        await getTags();
+        await getMembers();
+        await getStatuses();
```

Comment:
Fine for now, but I urge you to use `Promises.all()` to run these requests at the same time as opposed to sequentially.